superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -374,21 +375,21 @@ export function dbReducer(
       };
     case ActionType.setSSHTunnelLoginMethod:
       if (action.payload.login_method === AuthType.privateKey) {
-        const { password, ...rest } = trimmedState?.ssh_tunnel ?? {};
         return {
           ...trimmedState,
           ssh_tunnel: {
-            ...rest,
+            ...omit(trimmedState?.ssh_tunnel, ['password']),
```

Comment:
@AAfghahi It might be worth adding a conditional check to skip cases where no ssh tunnel info exists and consolidating the omit to a single check like:

```
case ActionType.setSSHTunnelLoginMethod:
  let ssh_tunnel;
  if (trimmedState?.ssh_tunnel) {
    // remove any attributes that are considered sensitive
    ssh_tunnel = omit(trimmedState.ssh_tunnel, [
      'password'
      'private_key',
      'private_key_password',
    ]);
  }

  return {
    ...trimmedState,
    ssh_tunnel,
  };
```
This way if additional fields get added at some point that also need to be filtered you can just add to the array of fields passed to omit without having more if blocks to differentiate between the AuthType.  omit will work fine getting attributes that may or may not exist on trimmedState.ssh_tunnel so we can just keep a single array of sensitive attributes

Reply:
`omit` handles `undefined` and `null` values (will return an empty object) already. About the conditionals, we need the properties to be removed individually so they don't collide, i.e, we shouldn't delete them all at once. The use case for this action is:

`If you select login method === password -> remove any private_key & private_key_password from the tunnel object we send in the payload to the API and vice versa.`

That way the tunnel always has only 1 login method and not the two of them which will cause issues later when trying to connect.

Reply:
Eric's method of making sure that there is no issue around null and undefined makes me a bit more comfortable, personally. 

Instead of omitting, could we use a `pick` here instead? @Antonio-RiveroMartnez is there a place where we define what columns the object should have? Are they the same btwn the two login methods?

Reply:
why would we pick `password`/`private_key`/`private_key_password` aren't those the ones we don't want included?

Reply:
The tunnel must have a login method, only one between `password` or `private_key`. If you select `password` as login method, you remove the `private_key`, and vice versa. So, before with the omit, if `login method === password` you `omit` the `private_key` and if you select `login method === private_key` you `omit` the `password`, that way your tunnel has only 1 login credentials.

Now, you want to use `pick`, in that case if you select `login method === password` you need to `pick` the common properties and include the `password`, otherwise there's no login credentials associated to the tunnel. Same if you select `login method === private_key`, you need to pick the common properties and include the `private_key` credentials.